crates/conversions/src/felt252.rs

```diff
@@ -120,35 +120,46 @@ impl TryInferFormat for Felt252 {
     }
 }
 
-pub trait SerializeAsFelt252Vec {
-    fn serialize_as_felt252_vec(&self) -> Vec<Felt252>;
+pub trait SerializeAsFelt252Vec: Sized {
+    fn serialize_as_felt252(self, output: &mut Vec<Felt252>);
```

Comment:
Why the name change here? It seems now that we're serializing down to a single felt (which is not true).